plugins/modules/aci_interface_policy_pim.py

```diff
@@ -0,0 +1,445 @@
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+        name_alias=dict(type="str"),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["pim", "tenant"]],
+            ["state", "present", ["pim", "tenant"]],
+        ],
+    )
+
+    aci = ACIModule(module)
+
+    tenant = module.params.get("tenant")
+    pim = module.params.get("pim")
+    authentication_key = module.params.get("authentication_key")
+    authentication_type = MATCH_PIM_INTERFACE_POLICY_AUTHENTICATION_TYPE_MAPPING.get(module.params.get("authentication_type"))
+    description = module.params.get("description")
+    name_alias = module.params.get("name_alias")
+    state = module.params.get("state")
+
+    designed_router_delay = module.params.get("designed_router_delay")
+    if designed_router_delay is not None and designed_router_delay not in range(1, 65536):
+        module.fail_json(msg="Parameter 'designed_router_delay' is only valid in range between 1 and 65535.")
+
+    designed_router_priority = module.params.get("designed_router_priority")
+    if designed_router_priority is not None and designed_router_priority not in range(1, 4294967296):
+        module.fail_json(msg="Parameter 'designed_router_priority' is only valid in range between 1 and 4294967295.")
+
+    hello_interval = module.params.get("hello_interval")
+    if hello_interval is not None and hello_interval not in range(1, 18724287):
+        module.fail_json(msg="Parameter 'hello_interval' is only valid in range between 1 and 18724286.")
+
+    join_prune_interval = module.params.get("join_prune_interval")
+    if join_prune_interval is not None and join_prune_interval not in range(60, 65521):
+        module.fail_json(msg="Parameter 'join_prune_interval' is only valid in range between 60 and 65520.")
+
+    if module.params.get("control_state"):
+        control_state = ",".join([MATCH_PIM_INTERFACE_POLICY_CONTROL_STATE_MAPPING.get(v) for v in module.params.get("control_state")])
+    else:
+        control_state = None
+
+    child_classes = dict(
+        pimJPInbFilterPol=module.params.get("inbound_join_prune_filter_policy"),
+        pimJPOutbFilterPol=module.params.get("outbound_join_prune_filter_policy"),
+        pimNbrFilterPol=module.params.get("neighbor_filter_policy"),
+    )
+
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="pimIfPol",
+            aci_rn="pimifpol-{0}".format(pim),
+            module_object=pim,
+            target_filter={"name": pim},
+        ),
+        child_classes=list(child_classes.keys()),
+    )
+
+    aci.get_existing()
+
+    if state == "present":
```

Comment:
I may be missing something but will this work if you're adding class_input for the first time?

Reply:
what do you mean @shrsr ?

Reply:
Ah nvm, I misunderstood the code.